lib/collection/src/shards/resolve.rs

```diff
@@ -0,0 +1,526 @@
+use std::collections::{HashMap, HashSet};
+use std::hash;
+
+use segment::types::ScoredPoint;
+use tinyvec::TinyVec;
+
+use crate::operations::types::Record;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum ResolveCondition {
+    All,
+    Majority,
+}
+
+pub trait Resolve: Sized {
+    fn resolve(responses: Vec<Self>, condition: ResolveCondition) -> Self;
+}
+
+impl Resolve for Vec<Record> {
+    fn resolve(records: Vec<Self>, condition: ResolveCondition) -> Self {
+        let mut resolved = Resolver::resolve(records, |record| record.id, PartialEq::eq, condition);
+        resolved.sort_unstable_by_key(|record| record.id);
+        resolved
+    }
+}
+
+impl Resolve for Vec<Vec<ScoredPoint>> {
+    fn resolve(batches: Vec<Self>, condition: ResolveCondition) -> Self {
+        // batches: <replica_id, <batch_id, ScoredPoint>>
+        // transpose to <batch_id, <replica_id, ScoredPoint>>
+
+        let batches = transpose(batches);
+
+        batches
+            .into_iter()
+            .map(|points| {
+                let mut resolved =
+                    Resolver::resolve(points, |point| point.id, scored_points_eq, condition);
+
+                resolved.sort_unstable();
```

Comment:
@generall Should we sort by `ScoredPoint::score` here? 🤔

Reply:
Ord is defined as:
```impl Ord for ScoredPoint {
    fn cmp(&self, other: &Self) -> Ordering {
        OrderedFloat(self.score).cmp(&OrderedFloat(other.score))
    }
}```

Reply:
Ok then.

Also, can't check `OrderedFloat` RN, but there's `f32::total_cmp`, which 1) is "built-in", 2) may be more complete than `OrderedFloat` impl.